### PR TITLE
Raise informative error on arg-reduction on unknown chunksize

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes https://github.com/dask/dask/issues/3108

- [x] Tests added / passed
- [x] Passes `flake8 dask`

### CODE REVIEW

File: dask/array/tests/test_reductions.py [hunk 1]
```diff
@@ -183,6 +183,18 @@ def test_nanarg_reductions(dfunc, func):
             dfunc(a).compute()
 
 
+@pytest.mark.parametrize('func', ['argmax', 'nanargmax'])
```
Comment (jrbourbeau) [inline dask/array/tests/test_reductions.py]: Do we want to add `argmin` and `nanargmin` here?

**Comment by mrocklin** [comment]
They use the same codepath.  I don't think it's necessary.  I tend to avoid
being super-exhaustive in tests, just because it adds to total testing time.

**Comment by jrbourbeau** [comment]
That's reasonable. Okay to merge, or did you want to add anything else?

**Comment by mrocklin** [comment]
It's ok by me.

**Pull request merged**